### DISCUSSION TITLE
v0.9.0-alpha4

### DISCUSSION THREAD

**Discussion by BenjamenMeyer** [vegastrike/Vega-Strike-Engine-Source#836] [Announcements]
## What's Changed
* Enhancement: Mac/Win Release processes by @BenjamenMeyer in https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/834

**Full Changelog**: https://github.com/vegastrike/Vega-Strike-Engine-Source/compare/v0.9.0-alpha3...v0.9.0-alpha4

<hr /><em>This discussion was created from the release <a href='https://github.com/vegastrike/Vega-Strike-Engine-Source/releases/tag/v0.9.0-alpha4'>v0.9.0-alpha4</a>.</em>

**Comment by stephengtuggy** [comment]
So, um, I actually never finished implementing the game_config refactor. There are still settings that the game engine doesn't read from disk properly. Quite a few of them, in fact. ☺️ 

Obviously, this is kind of a high priority. I plan to start working on it again soon. If anyone else wants to help out, that would also be welcome.

I think we should get this functionality finished up / fixed before releasing v0.9.0. What do you guys think?

**Reply by BenjamenMeyer** [reply]
@stephengtuggy agree - sounds important.

**Reply by BenjamenMeyer** [reply]
let's make sure there's a story for it on the 0.9.x board

**Reply by stephengtuggy** [reply]
https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/840

**Reply by royfalk** [reply]
Not sure what the status of this is, but think whether this is worth it. For master, we've moved to config.json and will also include auto-generated, removing most of the work. See https://github.com/royfalk/JsonStruct.

**Reply by stephengtuggy** [reply]
Hmm. Interesting. I'll check that out.

#840 was filed months ago. Realistically, I don't think it's feasible for us/me to complete in the 0.9.x time frame. We should probably just proceed without it.

**Comment by stephengtuggy** [comment]
@BenjamenMeyer the release process enhancements are looking good. Looks like you caught quite a few bugs that I had missed. Nice work.

**Reply by BenjamenMeyer** [reply]
Thanks...just need to figure out how to upload artifacts for Mac/Windows now. That's really all that's left.

**Reply by stephengtuggy** [reply]
@BenjamenMeyer how are we looking with regard to the release process now? I included a relevant change in #932 . I'm eager to see if we end up with a macOS installer as part of the next alpha release, as a result of that change.

**Reply by BenjamenMeyer** [reply]
We're close. I haven't had time to figure out picking up assets for Mac or Windows. That's the primary hold up for 0.9.x though iirc there were a few bugs that folks wanted to fix first. Not sure where we are on those.

**Reply by stephengtuggy** [reply]
@BenjamenMeyer Assets? As in `Assets-Production`? Or artifacts, as in the installers for the Vega Strike Game Engine? #932 should help with the latter. At least, that's my hope.

**Reply by stephengtuggy** [reply]
I added a call to CPack, which I think is the main ingredient that was missing from our CI process for generating releases, as far as Windows and Mac.

**Reply by BenjamenMeyer** [reply]
The artifact uploaders to get the binary/package artifacts from the builders....sorry, used the wrong term earlier. What we were using earlier only supported Linux, and I wasn't sure about using the default GitHub one. I think we migrated to one now that supports all platforms, but I don't know that we've tested it. Probably time to cut another Alpha once we get the Mac builds working again. Once we get all the artifacts uploading and correct we can move to beta.

**Reply by stephengtuggy** [reply]
OK. I enabled the macOS builds, and ran them manually for 0.9.x [here](https://github.com/vegastrike/Vega-Strike-Engine-Source/actions/runs/12451023157). All 4 builds appeared to succeed.

**Reply by stephengtuggy** [reply]
Mind if I kick off another Alpha of 0.9.x?

**Reply by stephengtuggy** [reply]
We have our first Mac installer! See https://github.com/vegastrike/Vega-Strike-Engine-Source/releases/tag/v0.9.0-alpha9

**Reply by stephengtuggy** [reply]
And now our first Windows installers! See https://github.com/vegastrike/Vega-Strike-Engine-Source/releases/tag/v0.9.0-alpha11
